jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/auth/IamAuthWidget.kt

```diff
@@ -16,48 +15,84 @@ import software.aws.toolkits.jetbrains.ui.AwsAuthWidget
 import software.aws.toolkits.resources.message
 import javax.swing.JPanel
 
-const val INSTANCE_ID_PROPERTY = "AWS.RdsInstanceId"
+const val DATABASE_HOST_PROPERTY = "AWS.RdsHost"
+const val DATABASE_PORT_PROPERTY = "AWS.RdsPort"
 
 class IamAuthWidget : AwsAuthWidget() {
-    private val instanceIdTextField = JBTextField()
+    private val databaseHostTextField = JBTextField()
+    private val databasePortTextField = JBTextField()
+
+    // Cache the most recently set URL and port, so if we save, and have
+    // emtpy text, we set HOST and PORT properties to the current values instead of
+    // having to split the datasource url string
+    private var cachedUrl: String = ""
+    private var cachedPort: String = ""
```

Comment:
I am confused on this, if it was set and cleared out we keep using it old value?

Reply:
Yes, but this is updated on every keystroke, we don't actually need it. We can pull the port and url off of the localdatasource, but I found doing that seemed weirder to me. I think I'll go back to doing that